src/enigma2/ConnectionManager.cpp

```diff
@@ -0,0 +1,180 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, write to
+ *  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ *  http://www.gnu.org/copyleft/gpl.html
+ *
+ */
+
+#include "ConnectionManager.h"
+
+// extern "C"
+// {
+// #include "libhts/htsmsg_binary.h"
+// #include "libhts/sha1.h"
+// }
+
+#include "p8-platform/os.h"
+#include "p8-platform/util/StringUtils.h"
+
+#include "../client.h"
+#include "IConnectionListener.h"
+#include "Settings.h"
+#include "utilities/Logger.h"
+#include "utilities/WebUtils.h"
+
+using namespace P8PLATFORM;
+using namespace enigma2;
+using namespace enigma2::utilities;
+
+#define FAST_RECONNECT_ATTEMPTS     (5)
+#define FAST_RECONNECT_INTERVAL   (500) // ms
+#define SLOW_RECONNECT_INTERVAL  (5000) // ms
+
+/*
+ * HTSP Connection handler
+ */
+
+ConnectionManager::ConnectionManager (IConnectionListener& connectionListener)
+  : m_connectionListener(connectionListener), m_suspended(false), m_state(PVR_CONNECTION_STATE_UNKNOWN)
+{
+}
+
+ConnectionManager::~ConnectionManager()
+{
+  StopThread(-1);
+  Disconnect();
+  StopThread(0);
+}
+
+void ConnectionManager::Start()
+{
+  // Note: "connecting" must only be set one time, before the very first connection attempt, not on every reconnect.
+  SetState(PVR_CONNECTION_STATE_CONNECTING);
+  CreateThread();
+}
+
+void ConnectionManager::Stop()
+{
+  StopThread(-1);
+  Disconnect();
+}
+
+void ConnectionManager::OnSleep()
+{
+  CLockObject lock(m_mutex);
+
+  Logger::Log(LogLevel::LEVEL_DEBUG, "%s going to sleep", __FUNCTION__);
+
+  m_suspended = true;
+}
+
+void ConnectionManager::OnWake()
+{
+  CLockObject lock(m_mutex);
+
+  Logger::Log(LogLevel::LEVEL_DEBUG, "%s Waking up", __FUNCTION__);
+
+  m_suspended = false;
+}
+
+void ConnectionManager::SetState ( PVR_CONNECTION_STATE state )
+{
+  PVR_CONNECTION_STATE prevState(PVR_CONNECTION_STATE_UNKNOWN);
+  PVR_CONNECTION_STATE newState(PVR_CONNECTION_STATE_UNKNOWN);
+
+  {
+    CLockObject lock(m_mutex);
+
+    /* No notification if no state change or while suspended. */
+    if (m_state != state && !m_suspended)
+    {
+      prevState = m_state;
+      newState  = state;
+      m_state   = newState;
+
+      Logger::Log(LogLevel::LEVEL_DEBUG, "connection state change (%d -> %d)", prevState, newState);
+    }
+  }
+
+  if (prevState != newState)
+  {
+    static std::string serverString;
+
+    if (newState == PVR_CONNECTION_STATE_SERVER_UNREACHABLE)
+    {
+      m_connectionListener.ConnectionLost();
+    }
+    else if (newState == PVR_CONNECTION_STATE_CONNECTED)
+    {
+      m_connectionListener.ConnectionEstablished();
+    }
+
+    /* Notify connection state change (callback!) */
```

Comment:
I see "my" code here... Seems, I coded something useful...

Reply:
Don't you always ;)